backend/scripts/curation/utils/ontology.py

```diff
@@ -0,0 +1,75 @@
+"""Methods for working with ontologies and the OLS."""
+from urllib.parse import quote_plus
+
+import requests
+
+OLS_API_ROOT = "http://www.ebi.ac.uk/ols/api"
+
+# Curie means something like CL:0000001
+
+
+def _ontology_name(curie):
+    """Get the name of the ontology from the curie, CL or UBERON for example."""
+    return curie.split(":")[0]
+
+
+def _ontology_value(curie):
+    """Get the id component of the curie, 0000001 from CL:0000001 for example."""
+    return curie.split(":")[1]
+
+
+def _double_encode(url):
+    """Double url encode a url. This is required by the OLS API."""
+    return quote_plus(quote_plus(url))
+
+
+def _iri(curie):
+    """Get the iri from a curie. This is a bit hopeful that they all map to purl.obolibrary.org"""
+    if _ontology_name(curie) == "EFO":
+        return f"http://www.ebi.ac.uk/efo/EFO_{_ontology_value(curie)}"
+    return f"http://purl.obolibrary.org/obo/{_ontology_name(curie)}_{_ontology_value(curie)}"
+
+
+class OntologyLookupError(Exception):
+    """Exception for some problem with looking up ontology information."""
+
+
+def get_ontology_label(curie):
+    """For a given curie like 'CL:1000413', get the label like 'endothelial cell of artery'"""
+
+    # If the curie is empty, just return an empty string. This happens wher there is no
+    # valid ontology value.
+    if not curie:
+        return ""
+    url = f"{OLS_API_ROOT}/ontologies/{_ontology_name(curie)}/terms/{_double_encode(_iri(curie))}"
+    response = requests.get(url)
```

Comment:
I gave this a whirl and it seems like this is gonna be super slow? I'm averaging about 1 second per lookup but if you have tens of thousands or hundreds of thousands of cells to look up, then this is gonna take a long time? 

One option could potentially be to use the owl files? Not fun, but it could be doable?

This could also just be a performance improvement we look into later on.

Reply:
This is meant to be used to create a dictionary mapping a set of unique CL terms to string labels. Then there should only be a handful of cell types that need to be looked up (even 10's of lookups is manageable) and then pandas `.map()` is very fast.

Reply:
Ohh I see; sounds reasonable then! I was imagining it being used for every cell label.